ls1intum/Artemis
# `Communication`: Speedup messaging

### Checklist
#### General


#### Server


### Motivation and Context


The query for retrieving recipients for WebSocket messages takes too much time on production systems.

### Description

The query does not return the whole user anymore. Additionally, the join with the course table has been removed.

### Steps for Testing

Prerequisites:
- 2 Students
- 1 Course with messaging enabled

1. Write messages in the on the messaging page, a lecture details page, and an exercise details page as student 1 in channels, group chats and direct chats that both students are a member of
2. Student 1 should see the message without much delay
3. Student 2 should see a notification (if not on the same page as student 1) or the message without much delay if they have opened the conversation. 
4. Student 2 does not receive duplicate notifications if mentioned
5. If student 1 writes in a channel or group chat that student 2 isn't a member of, student 2 does not receive notifications

### Review Progress






#### Performance Review
#### Code Review
#### Manual Tests

### Test Coverage







### Screenshots


No UI Changes
